wiki/concepts/qsl-qfapi.md

```diff
@@ -1,4 +1,40 @@
 ---
 title: QSL and QFAPI Overview
-draft: true
----
\ No newline at end of file
+---
+
+# QSL and QFAPI Overview
+
+This article explains the different standard libraries you might encounter when modding with Quilt.
+
+## What is an API?
+
+An API (Application Programming Interface) provides an interface for programs to interact with other programs. In the context of Minecraft modding, APIs tend to provide tools for better compatibility between mods or provide other useful extensions. For example, many of the bigger mods you might know of have an API which allows other mods to improve compatibility with them.
+
+In addition to that kind of API, there are libraries. Libraries themselves do not add any content to the game, and instead implement an API to make certain features simpler to implement for other developers. Fabric API, QSL and QFAPI are all official libraries published by the specific loader.
+
+It is important to note that an API only defines the *means* to interact with a specific program and the *corresponding behavior*, *not its actual implementation*. **The implementations may change at any time and thus shouldn't be used.** Implementations tend to contain `impl` in the package or class name, while APIs sometimes are in an `api` package.
+
+## Fabric API
+
+Fabric API (FAPI for short) is Fabric's API and provides some useful APIs that QSL does not provide. For Quilt, [Quilted Fabric API](#quilted-fabric-api) is an alternate implementation, that uses QSL where possible.
+
+## Quilt Standard Libraries
+
+The Quilt Standard Libraries (QSL for short) provide lots of core functionality for mods, as well as many useful APIs, such as [REA](../data/rea).
+
+The [main entrypoints](sideness#on-mod-initializers) are provided by QSL, and QSL loads the `assets` directory of your mod. This means that unless you use [mixin](mixins), basically nothing will work without QSL.
```

Comment:
```suggestion
Some of the features QSL provides are:
- The [main entrypoints](sideness#on-mod-initializers) for mods.
- Loading the `assets`  and `data` directory of your mod through the [resource loader](resource_loader).
- Many [events](events) such as world tick events.
- Extension classes and interfaces for many blocks, items, entities, and more.

While you can make your mod work with just [mixin](mixins), QSL does a lot of the heavy lifting and makes sure that common features mods have won't cause conflicts between them. QSL will also try to keep its API fairly stable between Minecraft versions, so that you have to do less work when updating your mod.
```